app/javascript/Pages/Home.jsx

```diff
@@ -1,37 +1,208 @@
+  const dif = ["beginner", "intermediate", "advanced"];
+
+  const fetchClassesDifficulty = () => {
+    let newClassData = [];
+    axios
+      .get(`/api/classes?difficulty=${difFilter}`)
+      .then((response) => {
+        newClassData = response.data;
+        setItems(newClassData);
+      })
+      .catch((error) => {
+        console.log(error);
+        newClassData = [];
+      });
+  };
+
+  const fetchCategories = async () => {
+    let categoryData = [];
+    axios
+      .get("/api/skills")
+      .then((response) => {
+        categoryData = response.data;
+        setCat(categoryData);
+      })
+      .catch((error) => {
+        console.log(error);
+        categoryData = [];
+      });
+  };
+
+  const fetchPopularClasses = async () => {
+    let popularClassData = [];
+    axios
+      .get("/api/classes")
+      .then((response) => {
+        popularClassData = response.data;
+        setPClasses(popularClassData);
+      })
+      .catch((error) => {
+        console.log(error);
+        popularClassData = [];
+      });
+  };
+
+  const addCategoryButtons = async () => {
+    let categories = [];
+    axios
+      .get(`/api/skills`)
+      .then((response) => {
+        categories = response.data;
+
+        setCategories(categories);
+      })
+      .catch((error) => {
+        console.log(error);
+        categories = [];
+      });
+  };
+
+  const fetchClasses = async () => {
+    const params = {};
+
+    if (categoryFilter) {
+      params.category_id = categoryFilter;
+    }
+
+    let url = "/api/classes";
+    let classes = [];
+    axios
+      .get(url, { params: params })
+      .then((response) => {
+        classes = response.data;
+
+        setPClasses(classes);
+      })
+      .catch((error) => {
+        classes = [];
+      });
+  };
+
+  const updateCategory = (category) => {
+    if (category) setCategoryFilter(category);
+    else {
+      setCategoryFilter("");
+    }
+  };
+  useEffect(() => {
+    fetchClasses();
+  }, [categoryFilter]);
+
+  const updateDif = (difficulty) => {
+    if (difficulty) setDifFilter(difficulty);
+    else {
+      setDifFilter("");
+    }
+  };
+
+  useEffect(() => {
+    fetchClassesDifficulty();
+  }, [difFilter]);
+
+  useEffect(() => {
+    fetchCategories();
+    fetchPopularClasses();
+    addCategoryButtons();
+  }, []);
+
+  return (
+    <main>
+      <Head title="Welcome" />
+      <Navbar current_user={props.current_user} />
+      <div className="hero-img">
+        <h1 className="hero-title">{t("hero-title")}</h1>
+      </div>
+      <div className="new-classes">
+        <h1 className="home-title">{t("first-title")}</h1>
+
+        <div className="filter-wrapper">
+          {dif.map((d, index) => {
+            return (
+              <ButtonFilter
+                key={index}
+                onClick={updateDif}
+                params={d}
+                id={difFilter}
+              >
+                {d.charAt(0).toUpperCase() + d.slice(1)}
```

Comment:
**suggestion (non-blocking)**: this would be better done in css with `text-transform: capitalize`

Reply:
But since is just the first letter and I give the text proprieties to the class, isn't more easy this way?

Reply:
with css is cleaner and more performant, but it's not a big issue